DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java

```diff
@@ -246,6 +258,33 @@ public void testSetKeywordWrongType() throws ClientProtocolException,
         assertEquals("Operation did not fail.", "error", status.asText());
     }
 
+    @Test
+    public void testConcurrentCalls() throws InterruptedException,
+            ExecutionException {
+        final ConnectionRequester connectionRequester = new ConnectionRequester();
+
+        Callable<String> task = new Callable<String>() {
+            @Override
+            public String call() throws Exception {
+                return connectionRequester.requestConnection();
+            }
+        };
+
+        List<Callable<String>> tasks = Collections.nCopies(numberOfThreads,
+                task);
+        ExecutorService executorService = Executors
+                .newFixedThreadPool(numberOfThreads);
+
+        List<Future<String>> futures = executorService.invokeAll(tasks);
+        List<String> resultList = new ArrayList<String>(futures.size());
```

Comment:
Let's go ahead and mark everything `final` that can be final.  Also the explicit repeated type is not necessary as of Java 7.

````java
final List<String> resultList = new ArrayList<>(futures.size());
````